src/postgres/src/backend/access/yb_access/yb_scan.c

```diff
@@ -2316,7 +2316,7 @@ YbBindHashKeys(YbScanDesc ybScan)
 		Assert(YbIsHashCodeSearch(key));
 		YbBound bound = {
 			.type = YB_YQL_BOUND_VALID,
-			.value = key->sk_argument
+			.value = (key->sk_argument > INT_MAX) ? 0 : key->sk_argument
```

Comment:
What about the case where it's below `INT_MIN`?
Why is this `>` and not `>=`?
Is there a test case against `key->sk_argument` being equal to `INT_MAX`?

Reply:
>As of now, I have only added the test cases present on the bug as you can see in file [yb_hash_code.sql] , Should I add the case where the user is passing INT_MAX as input instead of a negative integer?

Yes, please add more test cases as requested.

Reply:
@tanujnay112 INT_MAX is not accepted by yugabyte sql command line and since this value can be different depending on whether machine is 32 bit or 64 , it can not be hard coded in test cases. Can you suggest some way to integrate this test in here?
Also I tested locally with yb_hash_code(i) > 2^31 - 1 (INT_MAX for my machine) and yb_hash_code(i) >  2^31 (INT_MAX+1) and they both resulted in 0 rows while yb_hash_code(i) > -1 resulted in 2 rows. Since lc->data.int_value is of type signed int , when value bigger than INT_MAX is passed , it is not read as it is but only 32 bits are read.

Reply:
Yes please add those as test cases.

Reply:
@tanujnay112 Added the tests as suggested.

Reply:
Hi @parkhi, I've been looking into other yb_hash_code-related issue, basically, `yb_hash_code(hkey) < 65536` was reported not working properly.
I triaged the report and went to GitHub to check if it was reported before and found this work in progress.
Your fix is in right place, but you are missing a few points. 
First of all, the yb_hash_code returns a value from the range [0..65535]. If compared to an out of range constant, it is either "all" or "nothing", depending on the operation and whether the constant is above or below the range. 
It seems convenient to have out of bound condition checked when bound is applied to the total range in the YbApplyStartBound and YbApplyEndBound functions. The "Start" and "End" there defines the side, and out of bound value won't change the range, the function would just return true or false if the condition is respectively "all" or "nothing".

Second point (minor) is that uint64_t is not the best data type for the YbBound.value. It basically takes the value from the query, which is signed. The "long" value would allow typecast without data loss.